src/crypto/CHIPCryptoPAL.h

```diff
@@ -512,6 +513,54 @@ class AesCcm128Key
     uint8_t bytes[kAES_CCM128_Key_Length];
 };
 
+using AesKeyByteArray = uint8_t[CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES];
+
+/**
+ * @brief Platform-specific AES key
+ *
+ * The class represents AES key used by Matter stack (currently, only 128-bit key is supported by
+ * Matter) either in the form of raw key material or key reference, depending on the platform. To
+ * achieve that, it contains an opaque context that can be cast to a specific structure used by
+ * the given platform.
+ */
+class AesKey
+{
+public:
+    AesKey();
+    ~AesKey();
+
+    AesKey(const AesKey &) = delete;
+    AesKey(AesKey &&)      = delete;
+    void operator=(const AesKey &) = delete;
+    void operator=(AesKey &&) = delete;
+
+    CHIP_ERROR Initialize(const AesKeyByteArray & rawKey);
+
+    /**
+     * @brief Get internal context cast to the desired key representation
+     */
+    template <class T>
+    const T & Get() const
+    {
+        return *SafePointerCast<const T *>(&mContext);
+    }
+
+    /**
+     * @brief Get internal context cast to the desired key representation
+     */
+    template <class T>
+    T & Get()
```

Comment:
Why is this not const?

Reply:
Because we return a non-const reference pointing to bytes of the current object.

Reply:
You cannot overload on just constness. Please rename this get to `GetAsMutable` or such. Other classes in this module use `Bytes()` and `ConstBytes()`, but here it's not bytes.

Reply:
Why can't I? Most STL containers overload `begin()/end()` by constness. Or you mean that I shouldn't for some reason even though I can?

Reply:
It's because it can become ambiguous in some situations (e.g. `auto x = vec.begin();` versus `auto x = vec.cbegin()`)  both in terms of usage, and in terms of showing intent to the client. It becomes more important when there are more usecases of new abstracted classes, where a single method name causes more usage of the mutable rather than immutable version, since you have to go more out of your way to declare you want immutable, via the types assigned, or the signatures at play.

In STL, even though you have a const `begin`, there is the explicit `cbegin`. See also `FabricTable::GetMutableFabricByIndex` for the only way to get a fabric instance by index (and it's internal).